sql: TestShowLastQueryStatistics failed

sql.TestShowLastQueryStatistics [failed](https://mesolite.cluster.engflow.com/invocations/default/3432dc90-5a9d-4c16-af56-785d9ce10b85?testReportRun=1&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [a76eb5de5d3fdeb4f1276eefe5fdb11e6ebe53a4](https://github.com/cockroachdb/cockroach/commits/a76eb5de5d3fdeb4f1276eefe5fdb11e6ebe53a4):


```
=== RUN   TestShowLastQueryStatistics
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestShowLastQueryStatistics737936728
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:180: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:658: DRPC is enabled (via override by TestingGlobalDRPCOption)
    conn_executor_test.go:1507: unexpected plan latency: 1.123503
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestShowLastQueryStatistics737936728
--- FAIL: TestShowLastQueryStatistics (8.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowLastQueryStatistics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57528

sql.TestShowLastQueryStatistics [failed](https://mesolite.cluster.engflow.com/invocations/default/b4a058db-89d2-451f-999c-ce3d45ddc4d3?testReportRun=2&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [d36cdb971710d69d43c9f242d45019ddea71049b](https://github.com/cockroachdb/cockroach/commits/d36cdb971710d69d43c9f242d45019ddea71049b):


```
=== RUN   TestShowLastQueryStatistics
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestShowLastQueryStatistics1279649309
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:180: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:658: DRPC is enabled (via override by TestingGlobalDRPCOption)
    conn_executor_test.go:1507: unexpected plan latency: 2.564471
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestShowLastQueryStatistics1279649309
--- FAIL: TestShowLastQueryStatistics (12.18s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowLastQueryStatistics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestShowLastQueryStatistics [failed](https://mesolite.cluster.engflow.com/invocations/default/d4b44171-55a0-4d34-9ff2-9eb973d13713?testReportRun=1&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [9ecf4e87957bb1c5cd51e98590fc62160f43b9aa](https://github.com/cockroachdb/cockroach/commits/9ecf4e87957bb1c5cd51e98590fc62160f43b9aa):


```
=== RUN   TestShowLastQueryStatistics
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestShowLastQueryStatistics1107029085
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:182: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    conn_executor_test.go:1507: unexpected plan latency: 1.055451
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestShowLastQueryStatistics1107029085
--- FAIL: TestShowLastQueryStatistics (19.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #159480 sql: TestShowLastQueryStatistics failed [C-test-failure O-robot T-sql-queries branch-release-26.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowLastQueryStatistics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/conn_executor_test.go
+++ b/pkg/sql/conn_executor_test.go
@@ -1410,9 +1410,11 @@ func TestShowLastQueryStatistics(t *testing.T) {
 	ctx := context.Background()
 	var testTenant base.DefaultTestTenantOptions
 	if syncutil.DeadlockEnabled {
-		// Under deadlock, the planning latency in the external process mode can
-		// sometimes exceed 1s allowed limit.
-		testTenant = base.TestSkipForExternalProcessMode()
+		// Under deadlock, the planning latency in the secondary test tenant can
+		// sometimes exceed 1s allowed limit. (Note ideally we'd have used a
+		// different test tenant option since it's not "storage-layer specific"
+		// test, but there isn't a more suitable one.)
+		testTenant = base.TestIsSpecificToStorageLayerAndNeedsASystemTenant
 	}
 	s, sqlConn, _ := serverutils.StartServer(t, base.TestServerArgs{
 		DefaultTestTenant: testTenant,

